# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/mac/nsOSHelperAppService.mm
# Commit: d841771bed63
# Full Hash: d841771bed63e79bb6f61d18b64b9b60be29dff3
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2019-11-30 09:54:44
# Description:
#   Bug 1600351 - don't release things we don't own, r=barret
#   
#   Let's not be calling CFRelease on things we don't own.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55336
# ==============================================================================

diff -r ea930165f9f3 -r d841771bed63 uriloader/exthandler/mac/nsOSHelperAppService.mm
--- a/uriloader/exthandler/mac/nsOSHelperAppService.mm	Fri Nov 29 23:06:59 2019 +0000
+++ b/uriloader/exthandler/mac/nsOSHelperAppService.mm	Fri Nov 29 23:15:16 2019 +0000
@@ -181,7 +181,6 @@
       CFURLRef selfURL = ::CFBundleCopyBundleURL(appBundle);
       *_retval = ::CFEqual(selfURL, handlerBundleURL);
       rv = NS_OK;
-      ::CFRelease(appBundle);
       ::CFRelease(selfURL);
     }
     ::CFRelease(handlerBundleURL);
